blog/_posts/2016-10-06-julia-0.5-highlights.md

```diff
@@ -0,0 +1,717 @@
+
+    SortedDict([(k₁, v₁), (k₂, v₂)])        # fixed-size dictionaries
+    SortedDict([(k(i), v(i)) for i in c])   # dictionary comprehensions
+
+Not only are these constructions inconvenient and ugly, they're also inefficient since they create temporary heap-allocated arrays of heap-allocated tuples of key-value pairs.
+In Julia 0.5, with much relief, we can now instead write:
+
+    SortedDict(k₁ => v₁, k₂ => v₂)          # fixed-size dictionaries, since 0.4
+    SortedDict(k(i) => v(i) for i = c)      # dictionary comprehensions, since 0.5
+
+This syntax combines two orthogonal features introduced in 0.4 and 0.5, respectively: `k => v` as standalone syntax for a `Pair` object and generator expressions.
+The `Dict` type is now constructed in exactly the same way:
+
+    julia> Dict("foo" => 1, "bar" => 2)
+    Dict{String,Int64} with 2 entries:
+      "bar" => 2
+      "foo" => 1
+
+    julia> Dict("*"^k => k for k = 1:10)
+    Dict{String,Int64} with 10 entries:
+      "**********" => 10
+      "***"        => 3
+      "*******"    => 7
+      "********"   => 8
+      "*"          => 1
+      "**"         => 2
+      "****"       => 4
+      "*********"  => 9
+      "*****"      => 5
+      "******"     => 6
+
+This generalization makes the syntax for constructing a `Dict` slightly longer, but we feel that the increased consistency, ability to change dictionary implementations with a simple search-and-replace, and putting user-defined dictionary-like types on the same level as the standard `Dict` make this change well worthwhile.
+
+## Arrays
+
+The 0.5 release was originally intended to include a large number of disruptive array changes, collectively dubbed "Arraymageddon".
+After much discussion, experimentation and benchmarking, this set of breaking changes was significantly reduced for a variety of reasons:
+
+- Some changes were deemed not to be good ideas after all;
+- Others were of unclear benefit, so it was decided to reconsider them in the future once more evidence became available one way or the other;
+- A few didn't get implemented due to lack of developer time, including some cases where everyone agrees there's a problem but no complete design for a solution.
+
+Although most of the breaking changes didn't happen in 0.5, in a less obvious way, this was a major release for Julia's array infrastructure.
+The code to implement the various complex polymorphic indexing operations for generic arrays and array-like structures was majorly refactored, and in the process it shrank by 40% while becoming more complete, more general, and faster.
+You can read more about the very cool things you can do with array-like types in 0.5 in an excellent pair of blog posts published here earlier in the year on [multidimensional iteration tools](/blog/2016/02/iteration) and [generalizing abstract arrays](/blog/2016/03/arrays-iteration).
+
```

Comment:
Yeah, I didn't want to make too much of that since Julia's built-in arrays still index from 1 – it's just that the generic operations support arrays that index from other values. It would be great to have a separate blog post about it.